dhis-2/dhis-support/dhis-support-system/src/test/java/org/hisp/dhis/system/util/ValidationUtilsTest.java

```diff
@@ -493,6 +495,31 @@ void testValueIsValid() {
     assertNotNull(valueIsValid("2012304-01", de));
     assertNotNull(valueIsValid("Date", de));
 
+    de.setValueType(TIME);
+    assertNull(valueIsValid("12:30", de));
+    assertNotNull(valueIsValid("25:00", de));
+
+    de.setValueType(AGE);
+    assertNull(valueIsValid("2013-04-01", de));
+    assertNotNull(valueIsValid("2012304-01", de));
+    assertNotNull(valueIsValid("Date", de));
+
+    de.setValueType(PHONE_NUMBER);
+    assertNull(valueIsValid("(+47) 3398 7937", de));
+    assertNotNull(valueIsValid("12345", de));
+
+    de.setValueType(EMAIL);
+    assertNull(valueIsValid("enrico@dhis2.org", de));
```

Comment:
are you looking to get spam 😅

Reply:
You are right, lets go with something a little bit more generic :grimacing: 
```suggestion
    assertNull(valueIsValid("validemail@dhis2.org", de));
```